### PR TITLE
Refactor media to return IPublishedContent

### PR DESCRIPTION
**Pull request by stevetemple**

### CODE REVIEW

File: Gibe.DittoProcessors/Processors/AbsoluteUrlAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+using Umbraco.Web;
+
+namespace Gibe.DittoProcessors.Processors
+{
+	public class AbsoluteUrlAttribute : TestableDittoProcessorAttribute
+	{
+
+		public override object ProcessValue()
+		{
+			var content = Value as IPublishedContent;
```
Comment (aheaford) [inline Gibe.DittoProcessors/Processors/AbsoluteUrlAttribute.cs]: Perhaps `(Value as IPublishedContent) ?? Context.Content`? Though if we're making changes here, perhaps we could put something much like that into one of the supertypes here to make it more likely we can use that logic wherever valuable.

**Comment by aheaford** [comment]
Other than one minor processor tweak, LGTM. Always promising to see a changeset where the removed lines vastly outstrip the added lines!

File: README.md [hunk 2]
```diff
@@ -7,18 +7,22 @@ Install via nuget ``` install-package Gibe.DittoProcessors ```
 ## Processors
 | Processor | Description |
 |:----------|:------------|
+|AbsoluteUrlAttribute| Return the ```UrlAbsolute``` of an ```IPublishedContent``` |
```
Comment (stevetemple) [inline README.md]: Remove Attribute

File: Gibe.DittoProcessors/Processors/AbsoluteUrlAttribute.cs (hunk 1 above)
Comment (stevetemple) [inline Gibe.DittoProcessors/Processors/AbsoluteUrlAttribute.cs]: Think I might prefer attributes that are CurrentContentAsAbsoluteUrl naming or something like that for dealing with the Context.Content?

Comment (aheaford) [inline Gibe.DittoProcessors/Processors/AbsoluteUrlAttribute.cs]: Hm, what's the benefit of having two types which perform the same operation here? Any context in which one is valid, the other would never be valid, so it doesn't seem like we gain flexibility.

Comment (stevetemple) [inline Gibe.DittoProcessors/Processors/AbsoluteUrlAttribute.cs]: I think we need to be clearer where we have one that uses a supplied [UmbracoProperty("x")] and one that uses the current node

**Pull request merged**